components/NavBar/index.tsx

```diff
@@ -60,8 +60,9 @@ const Logo = (): JSX.Element => {
   );
 };
 
-const UserAvatar = ({ isOpen, avatar, toggleNavDropdown }) => {
-  const { currentUserBalance } = useAuthContext();
```

Comment:
We should invoke `updateCurrentUserBalance` in the purchase NFT function instead of in the `UserAvatar`.